src/mantarray_file_manager/files.py

```diff
@@ -474,24 +470,24 @@ def get_raw_reference_reading(self) -> NDArray[(2, Any), int]:
         return self._raw_ref_reading
 
     def _check_for_trimmed_file(
-        self, times: NDArray[(1, Any), int], time_delta_centimilliseconds: int
+        self, times: NDArray[int], time_delta_centimilliseconds: int
     ) -> int:
-        file = self.get_h5_file()
-        metadata_keys = set(file.attrs.keys())
-        if str(IS_FILE_ORIGINAL_UNTRIMMED_UUID) in metadata_keys:
+        try:
             is_untrimmed = self.get_h5_attribute(str(IS_FILE_ORIGINAL_UNTRIMMED_UUID))
             if not is_untrimmed:
                 time_trimmed = self.get_h5_attribute(
                     str(TRIMMED_TIME_FROM_ORIGINAL_START_UUID)
                 )
                 new_times = times + time_delta_centimilliseconds
-                start_index = _find_start_index(time_trimmed, new_times)
-                time_delta_centimilliseconds = new_times[start_index]
-                return time_delta_centimilliseconds
-        return 0
+                start_index = find_start_index(time_trimmed, new_times)
+                new_time_delta = int(new_times[start_index])
+                return new_time_delta
+        except FileAttributeNotFoundError:
```

Comment:
generally, you want to have a `try` statement wrap the minimum possible amount of code.  That way you know you are catching specifically the exception you think you are.  If you wrap more lines of code, then more and more errors could potentially be thrown by different lines of code in the `try` block, so there's a risk that some other error you didn't intend to catch gets thrown by a different line of code